### ISSUE TITLE
Consider moving the `$ErrorView` configuration to `$PSStyle`

### ISSUE THREAD

**Issue by daxian-dbw** [PowerShell/PowerShell#17917]
### Summary of the new feature / enhancement

The `$ErrorView` variable was introduced earlier than `$PSStyle`. But with `$PSStyle` containing configuration like `$PSStyle.Formatting` and `$PSStyle.Progress`, it feels to me the error view configuration should be moved to `$PSStyle` as well, for consistency and easier discoverability.

### Proposed technical implementation details (optional)

Discard the `$ErrorView` variable (breaking change), add the public property `ErrorView` to `System.Management.Automation.PSStyle`.

**Comment by iSazonov** [comment]
We could simplify codes if we moved $host.UI.SupportsVirtualTerminal detection in PSStyle too.

**Comment by daxian-dbw** [comment]
> We could simplify codes if we moved $host.UI.SupportsVirtualTerminal detection in PSStyle too.

I don't think we should do that because the engine is supposed to be agnostic to the host capabilities. Think about the scenario where PowerShell is hosted in a Windows GUI application.

**Comment by dkaszews** [comment]
Idea for softer transition without breaking change just yet: make `$ErrorView` getters and setters forward to `$PSStyle.ErrorView` with a deprecation warning, similarly how code libraries deprecate functions. Then we can fully remove it in another major release or next one. That way people who update regularly will get some time to adjust.

**Comment by daxian-dbw** [comment]
Engine Working Group review this today. We propose to move the error view information to `PSStyle`, and then have `ErrorView` variable point to that property within `PSStyle`, similarly to how `$ExecutionContext` points to an internal object in the PS engine.
After that change, we should mark the `$ErrorView` variable deprecated, but will keep it around for a long time to avoid breaking change.

**Comment by dkaszews** [comment]
I would like to work on this, but I have another PR https://github.com/PowerShell/PowerShell/pull/18003 that is modifying `$PSStyle` that has been reviewed but not merged for almost a year, @daxian-dbw can you please resolve it first so I don't merge-conflict myself?

**Comment by dkaszews** [comment]
Bump

**Comment by dkaszews** [comment]
Bump, still blocked by unmerged #18003 half a year later

@SteveL-MSFT I got an approval from you a while ago, could you please merge?

**Comment by dkaszews** [comment]
Bump, still blocked by unmerged #18003 half a year later

@SteveL-MSFT I got an approval from you a while ago, could you please merge?

**Comment by dkaszews** [comment]
Bump, still blocked by unmerged #18003 half a year later

@SteveL-MSFT I got an approval from you a while ago, could you please merge?
